### PR TITLE
Additional isolation of ASGS, moving to more self-containment.

### LINKED ISSUE #880: "isolated ASGS" installation improvements
Everything should be in the local `asgs` directory, including:

* profiles
* ./opt

Keep the following in `$HOME`:

* `~/asgs-global.conf`
* `~/.asgsh_profile`

TBD:

* ADCIRC src directories, `.adcirc-meta` directory

### PR DESCRIPTION
**Pull request by wwlwpd**
Issue 880: Mainly updating ./init-asgs.sh to use existing asgs-brew.pl
flags to specify --home; which means opt, tmp, and .asgs are in what
is effectively $SCRIPTDIR. Added a small script to add useful dev
tools for Perl optiona, (via "build perl-dev" command) and Perl PDL
via "build pdl" command

Found and fixed unrelated bug with PLATFORM_INIT not getting set in
the auto-generated, "update-asgs". The result was PLATFORM_INIT was
not getting written to the new "./asgsh" wrapper used to start ASGS'
Shell Environment.

Uses the following, relative to the asgs cloned directory:

./.adcirc-meta       # stores ADCIRC build info (for `load adcirc`)
./opt                # install path for most compiled things
./opt/models/adcircs # default build directory for ADCIRCs
./profiles           # profiles, was $HOME/.asgs
./tmp                # use as a temporary build directory for many things

Resolves #880.

### CODE REVIEW

File: init-asgs.sh [hunk 1]
```diff
@@ -164,9 +165,9 @@ case "$platform" in
        export PLATFORM_INIT=$(readlink -f "${_PLATFORM_INIT[$platform]}")
        source "$PLATFORM_INIT"
      else
-      echo "Can't find 'init.sh' for '$platform' or Unknown defaults for platform '$platform', using "$HOME" as 'WORK' and 'SCRATCH' directories..."
-      WORK=${WORK:-$HOME}
-      SCRATCH=${SCRATCH:-$HOME}
+      echo "Can't find 'init.sh' for '$platform' or Unknown defaults for platform '$platform', using "$ASGS_HOME" as 'WORK' and 'SCRATCH' directories..."
+      WORK=${WORK:-$ASGS_HOME}
+      SCRATCH=${SCRATCH:-$ASGS_HOME}
```
Comment (jasonfleming) [inline init-asgs.sh:170]: this may the issue with `$RUNDIR` ending up as `$ASGS_HOME` as we discussed

File: init-asgs.sh [hunk 2]
```diff
@@ -256,7 +258,7 @@ if [ -z "$profile" ]; then
 fi
 
 if [ -z "$BATCH" ]; then
-  if [ -e $HOME/.asgs/$profile ]; then
+  if [ -e $ASGS_HOME/.asgs/$profile ]; then
```
Comment (jasonfleming) [inline init-asgs.sh:261]: probably change `.asgs` to `profiles`

**Review by jasonfleming — COMMENTED** [review]
Hey @wwlwpd I know this is still a draft but I added comments as suggested in our call this am :-)

File: init-asgs.sh (hunk 1 above)
Comment (wwlwpd) [reply init-asgs.sh:170]: I think it's an artifact on my particular installation. I was able to show that it sets to what is exported based on the platform specific stuff that's set in `init-asgs.sh` or (preferably) in the platform's definition, like in `./platforms/mike/init.sh`. We'll just have to keep an eye on it.

File: init-asgs.sh [hunk 3]
```diff
@@ -267,7 +269,7 @@ if [ -z "$BATCH" ]; then
 fi
 
 
-_default_installpath=$WORK/opt
+_default_installpath=$(pwd)/opt
```
[the remark this answers is not in the document]
Comment (wwlwpd) [reply init-asgs.sh:272]: yes! good catch.

**Review by jasonfleming — APPROVED** [review]
This looks great @wwlwpd thank you. I am looking forward to working with this! :-)

**Pull request merged**